android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -565,6 +565,12 @@ public void setSource(WebView view, @Nullable ReadableMap source) {
                 view.getSettings().setUserAgentString(headers.getString(key));
               }
             } else {
+              if (key.equalsIgnoreCase("Cookie")) {
+                URI uri = new URI(url);
+                String domain = uri.getHost();
+                CookieManager.getInstance().setCookie(domain, headers.getString(key));
```

Comment:
According to the [documentation](https://github.com/react-native-webview/react-native-webview/blob/master/docs/Guide.md#managing-cookies), cookies added via the `source` `headers` prop "will only be sent on the first request"

Setting the cookies via `CookieManager` seems like it would affect more than just the first request?

Reply:
Good point but this feature didn't work at all without setting the cookie in the CookieManager. Plus, that note has a "set custom headers" link too, which seems to include a "workaround" to exactly this, implying that this is a problem already.

Maybe we should modify the docs to only include **iOS** to that statement? And say on android you have to clear the cookie if you don't want it to carry over to consecutive calls to that domain? But I would like direction on this, as I am not an Android guy :)

Reply:
@ccreonopoulos I get what you're saying, I just would rather not trade one confusing behavior for another. In any case it does seem like the documentation should be updated. That section of the guide in general seems to be iOS-specific, quite possibly written before Android support was even really a thing.

If the `Cookie` `headers` property is going to behave differently from the other `headers` properties, and not just apply to the initial `source` request, that suggests to me that it should be a separate property/method, not part of `source`.

Which then makes me wonder what the difference is between this approach and using [@react-native-community/cookies](https://github.com/react-native-community/cookies) as suggested in the guide?

Reply:
Thank you for your suggestions @TheAlmightyBob! If you think a separate property/method would be something we could accommodate, it would be much more preferable if this PR is not something you'd like to consider further. 

The reasons we didn't use the cookie library are:

1. We needed custom cookie usage, which although trying it the exact way the documentation suggested `If you'd like to send custom cookies in the WebView itself`, the "Cookie" header wasn't working (hence the the original reason for this PR - case sensitivity)

2. It would be a ton of overhead to add a new library and since our application is highly security related, we didn't want to have to increase our attack surface.

Reply:
> We needed custom cookie usage, which although trying it the exact way the documentation suggested `If you'd like to send custom cookies in the WebView itself...`

So, I didn't write that of course, but the way I interpret that documentation is by emphasizing the _in the WebView itself_ part, to set cookies for only a specific web request without adding them to a shared cookie manager. But your code here is just adding them to the cookie manager anyway... which is [the same thing react-native-cookies does](https://github.com/react-native-cookies/cookies/blob/master/android/src/main/java/com/reactnativecommunity/cookies/CookieManagerModule.java#L181)?

> a ton of overhead to add a new library

I understand that pain, but it's a slippery slope to just cramming every feature into a single giant library when specialized libraries for those features already exist.